lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py

```diff
@@ -0,0 +1,108 @@
+from sycamore.data import Element, BoundingBox
+from sycamore.utils.cache import DiskCache
+from io import IOBase
+from typing import BinaryIO, Tuple, List, Union, cast, Any
+from pathlib import Path
+from sycamore.utils.import_utils import requires_modules
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+pdf_miner_cache = DiskCache(str(Path.home() / ".sycamore/PDFMinerCache"))
+
+
+class PDFMinerExtractor:
+    @requires_modules(["pdfminer", "pdfminer.utils"], extra="local-inference")
+    def __init__(self):
+        from pdfminer.converter import PDFPageAggregator
+        from pdfminer.layout import LAParams
+        from pdfminer.pdfinterp import PDFPageInterpreter, PDFResourceManager
+
+        rm = PDFResourceManager()
+        param = LAParams()
+        self.device = PDFPageAggregator(rm, laparams=param)
+        self.interpreter = PDFPageInterpreter(rm, self.device)
+
+    def _open_pdfminer_pages_generator(self, fp: BinaryIO):
```

Comment:
Isn't this kind of redundant with pdf_to_pages?

Reply:
Removed.